src/io/flutter/run/bazelTest/BazelTestRunner.java

```diff
@@ -252,67 +254,64 @@ public BazelPositionMapper(@NotNull final Project project,
-      if (filePath.startsWith(runfilesDir) && workspaceOffset != -1) {
-        for (String workspaceLoc : workspaceLocations) {
-          final VirtualFile virtualFileInProject =
-            LocalFileSystem.getInstance().findFileByPath(workspaceLoc + filePath.substring(workspaceOffset, filePath.length()));
-          if (virtualFileInProject != null) {
-            return XDebuggerUtil.getInstance().createPosition(virtualFileInProject, xSourcePosition.getLine(), xSourcePosition.getOffset());
+      // Verify the returned workspace directory name, we weren't passed a workspace name or if the valid workspace name does not start the
+      // uri then return the super invocation of this method. This prevents the unknown URI type from being passed to the analysis server.
+      if (!verifyWorkspaceNameValue(workspaceDirName) || !uri.startsWith(workspaceDirName + ":/")) return super.findLocalFile(uri);
+
+      final String pathFromWorkspace = uri.substring(workspaceDirName.length() + 1, uri.length());
+
+      // For each root in each module, look for a bazel workspace path, if found attempt to compute the VirtualFile, return when found.
+      return ApplicationManager.getApplication().runReadAction((Computable<VirtualFile>)() -> {
+        for (Module module : DartSdkLibUtil.getModulesWithDartSdkEnabled(getProject())) {
+          for (ContentEntry contentEntry : ModuleRootManager.getInstance(module).getContentEntries()) {
+            final VirtualFile includedRoot = contentEntry.getFile();
+            if (includedRoot == null) continue;
+
+            final String includedRootPath = includedRoot.getPath();
+            final int workspaceOffset = includedRootPath.indexOf(workspaceDirName);
+            if (workspaceOffset == -1) continue;
+
+            final String pathToWorkspace = includedRootPath.substring(0, workspaceOffset + workspaceDirName.length());
+            final VirtualFile virtualFile = LocalFileSystem.getInstance().findFileByPath(pathToWorkspace + pathFromWorkspace);
+            if (virtualFile != null) {
+              return virtualFile;
+            }
           }
         }
-      }
-      return xSourcePosition;
+        return super.findLocalFile(uri);
+      });
     }
 
     /**
-     * Return true if the passed runfilesDir and workspaceDirName are not null, are not empty, and the runfilesDir ends with
-     * {@value RUNFILES_SLASH}.
+     * Return true if the passed workspaceDirName is not null and is not empty.
      */
-    private boolean verifyRunFilesAndWorkspaceNameValues(@Nullable final String runfilesDir, @Nullable final String workspaceDirName) {
-      return runfilesDir != null && runfilesDir.endsWith(RUNFILES_SLASH) && workspaceDirName != null && !workspaceDirName.isEmpty();
+    private boolean verifyWorkspaceNameValue(@Nullable final String workspaceDirName) {
+      return workspaceDirName != null && !workspaceDirName.isEmpty();
```

Comment:
Consider `!StringUtils.isEmpty(workspaceDirName)` ?  Also, maybe just inline that call and remove this method?